orthoinference/src/main/java/org/reactome/orthoinference/SkipInstanceChecker.java

```diff
@@ -101,9 +107,111 @@ public static boolean checkIfInstanceShouldBeSkipped(GKInstance reactionInst) th
+		// the same collection causes outputs to be inferred in both inputs and outputs during the actual inference -- not ideal
+		Collection<GKInstance> reactionInputs = reactionInst.getAttributeValuesList(input);
+		Collection<GKInstance> reactionOutputs = reactionInst.getAttributeValuesList(output);
+		Collection<GKInstance> reactionCatalystPEs = new ArrayList<>();
+		Collection<GKInstance> reactionCatalysts = reactionInst.getAttributeValuesList(catalystActivity);
+		for (GKInstance reactionCatalyst : reactionCatalysts) {
+			GKInstance catalystPE = (GKInstance) reactionCatalyst.getAttributeValue(physicalEntity);
+			if (catalystPE != null) {
+				reactionCatalystPEs.add(catalystPE);
+			}
+		}
+		// Screen inputs
+		for (GKInstance reactionInput : reactionInputs) {
+			if (!componentIsInferrable(reactionInput)) {
+				return false;
+			}
+		}
+		// Screen outputs
+		for (GKInstance reactionOutput : reactionOutputs) {
+			if (!componentIsInferrable(reactionOutput)) {
+				return false;
+			}
+		}
+		// Screen catalyst PhysicalEntities
+		for (GKInstance reactionCatalystPE : reactionCatalystPEs) {
+			if (!componentIsInferrable(reactionCatalystPE)) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	// This looks a lot like the code structure found in OrthologousEntityGenerator, just without the actual inference code or override functionality
+	// This will screen all instance types except for some EntitySets, which are complex to screen ahead of time.
+	private static boolean componentIsInferrable(GKInstance reactionComponent) throws Exception {
+		// This block doesn't do anything aside from prevent non-species-containing instances from going through other screening.
+		// During actual inference, a non-species-containing instance would be returned without any inference.
+		if (!SpeciesCheckUtility.checkForSpeciesAttribute(reactionComponent)) {
+//				return true;
+		} else if (reactionComponent.getSchemClass().isa(GenomeEncodedEntity))
+		{
+			if (reactionComponent.getSchemClass().toString().contains(EntityWithAccessionedSequence)) {
+				String referenceEntityId = ((GKInstance) reactionComponent.getAttributeValue(referenceEntity)).getAttributeValue(identifier).toString();
+				if (homologueMappings.get(referenceEntityId) == null) {
+					return false;
+				}
+			} else {
+				return false;
+			}
+		} else if (reactionComponent.getSchemClass().isa(Complex) || reactionComponent.getSchemClass().isa(Polymer) || reactionComponent.getSchemClass().isa(EntitySet)) {
+			List<Integer> complexProteinCounts = ProteinCountUtility.getDistinctProteinCounts(reactionComponent);
+			int totalProteinCounts = complexProteinCounts.get(0);
+			int inferrableProteinCounts = complexProteinCounts.get(1);
+			if (reactionComponent.getSchemClass().isa(Complex) || reactionComponent.getSchemClass().isa(Polymer)) {
+				int percent = 0;
+				if (totalProteinCounts > 0) {
+					percent = (inferrableProteinCounts * 100) / totalProteinCounts;
```

Comment:
Using an `int` for `percent` doesn't cause any loss-of-precision problems?

Reply:
`double percent = 0` at line 199 would be good